File: javascript/day02/part1-2.js
const fs = require('fs')
let input = fs.readFileSync('input', 'utf-8').trim()

function execute (noun, verb, memory){
  memory[1] = noun
  memory[2] = verb

  for(let i = 0; i < memory.length; i+=4){
    let opcode = memory[i]
    let first = memory[i+1]
    let second = memory[i+2]
    let third = memory[i+3]

    if(opcode == 99)
      break

    if(opcode == 1)
      memory[third] = memory[first] + memory[second]
    if(opcode == 2)
      memory[third] = memory[first] * memory[second]
  }

  return memory[0]
}


console.log(execute(12,2, input.split(',').map(Number)))

for(let i = 0; i < 100; i++){
  for(let j = 0; j < 100; j++){
    let r = execute(i, j, input.split(',').map(Number))
    if(r == 19690720)
      console.log(100*i+j)
  }
}

File: python/day02/part1-2.py
f = open('input', 'r')
f = f.readlines()[0].split(',')
f = [int(x) for x in f]

def execute(noun, verb, memory):
    memory[1] = noun
    memory[2] = verb

    for i in range(0, len(memory), 4):
        first = memory[i+1]
        second = memory[i+2]
        third = memory[i+3]

        if(memory[i] == 99):
            return memory[0]

        if(memory[i] == 1):
            memory[third] = memory[first] + memory[second]

        if(memory[i] == 2):
            memory[third] = memory[first] * memory[second]

    return memory[0]

print(execute(12,2,f[:]))

for i in range(0, 100):
    for j in range(0, 100):
        result = execute(i, j, f[:])
        if(result == 19690720):
            print(100*i+j)
